### PR TITLE
Adding periodic job for azuredisk in a capz + windows cluster

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

/assign @chewong @andyzhangx 
@jsturtevant @CecileRobertMichon - FYI

### CODE REVIEW

**Comment by andyzhangx** [comment]
`pull-kubernetes-e2e-capz-azure-disk` test on k/k master branch is still failing, need to fix that first, and then enable this job:
https://testgrid.k8s.io/provider-azure-presubmit#pull-kubernetes-e2e-capz-azure-disk

**Comment by marosset** [comment]
csi-proxy isn't getting installed on the nodes in `pull-kubernetes-e2e-capz-azure-disk` test on k/k master

https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/106678/pull-kubernetes-e2e-capz-azure-disk/1506727123799248896/artifacts/workload-cluster/capz-pxxhep-md-0-mt6gq/pods/

Not sure why this is yet.

File: config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml [hunk 1]
```diff
@@ -543,6 +543,64 @@ periodics:
...
+            memory: "4Gi"
+        env:
+          - name: WINDOWS
+            value: "true"
+          - name: TEST_WINDOWS
+            value: "true"
+          - name: WINDOWS_USE_HOST_PROCESS_CONTAINERS
+            value: "true"
+          - name: NODE_MACHINE_TYPE
+            value: "Standard_D4s_v3"
```
[the remark this answers is not in the document]
Comment (marosset) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:595]: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/ad1fb13d4fc39453aed16a7beffba64c40dff25c/scripts/ci-entrypoint.sh#L64 controls the number of worker machines.

I will make updates to set this to 0 and also specify a more recent K8s version

**Comment by marosset** [comment]
/close
in favor of https://github.com/kubernetes/test-infra/pull/25945/

**Pull request closed without merging**